### ISSUE TITLE
python tempfile-without-flush: Don't require .write() for exception?

### ISSUE THREAD

**Issue by The-Compiler** [returntocorp/sgrep-rules#1000]
**Describe the bug**
The `python.lang.correctness.tempfile.flush.tempfile-without-flush` rule has [exceptions](https://github.com/returntocorp/semgrep-rules/blob/35e9f350a5d0190c936502fff179ba2c465c4438/python/lang/correctness/tempfile/flush.yaml#L5-L20) if `.write()` and then `.close()` or `.flush()` is used on a tempfile. However, this doesn't cover code which calls `.close()` immediately, without writing:

```
$ semgrep --config=p/r2c-ci --verbose test.py
[...]
severity:error rule:python.lang.correctness.tempfile.flush.tempfile-without-flush: possibly missing a .flush() or .close() call to temporary file handle; file may or may not exist when handle.name is used
3:handle = tempfile.NamedTemporaryFile(delete=False)
4:handle.close()
5:print(handle.name)
```

**To Reproduce**
```python3
import tempfile

handle = tempfile.NamedTemporaryFile(delete=False)
handle.close()
print(handle.name)
```

**Expected behavior**
It seems to me like the exception should trigger whenever `.close()` is invoked, no matter if `.write()` is called before that or not?

**Comment by The-Compiler** [comment]
Oh, nice issue number I got there :smile: 

I'd be happy to submit a PR for this if others agree this is the way to go and I'm not missing anything - taking the freedom to ping @ievans who originally added the rule in 64922d467c20e802668a8608204c62449f286b5c.

**Comment by k-evans** [comment]
Very good point @The-Compiler, would be happy to accept a PR for that! cc @mschwager who also worked on the original rule

**Comment by ievans** [comment]
Very good point @The-Compiler, would be happy to accept a PR for that! cc @mschwager who also worked on the original rule
